packages/forgae-lib/forgae-deployer.ts

```diff
@@ -145,345 +149,42 @@ export class Deployer {
-        } else if (argument.list) {
-            let result = parseACIFunctionArgumentsList(argument.list);
-            return result;
-        }
-    }
-}
 
-function parseACIFunctionArgumentsList(list: any[]): any {
-    
-    let temp = [];
 
-    if (list.length === 1 && typeof list[0] === 'string') {
-        temp.push(list[0]);
-    } else {
-
-        for (let element of list) {
-            let result = _parseACIFunctionArguments(element);
-            temp.push(result);
-        }
     }
-
-    return `list(${ temp.toString() })`;
 }
 
-function parseACIFunctionArgumentsRecord(record): any {
-    let temp = [];
-    for (let value of record) {
-
-        if (value.type.length === 1 && typeof value.type[0] === 'string') {
-            temp.push(value.type);
-        } else {
-            let tempSubArr = [];
-            for (let arg of value.type) {
-                if (typeof arg === 'string') {
-                    tempSubArr.push(arg);
-                } else {
-
-                    let result = parseACIFunctionArgumentsRecord(arg.record);
-                    // remove brackets
-                    result = result.substr(1, result.length - 2);
-                    tempSubArr.push(result);
-                }
-            }
-
-            temp.push(`(${ tempSubArr.toString() })`);
-        }
+async function generateInstancesWithWallets(network: Network, contractAddress) {
+    instances = [];
+    for (let wallet in nodeConfig.defaultWallets) {
+        let currentClient: Client = await utils.getClient(network, nodeConfig.defaultWallets[wallet]);
+        let contractInstance: ContractInstance = await currentClient.getContractInstance(contract, { contractAddress });
+        instances.push(contractInstance)
+        instancesMap[nodeConfig.defaultWallets[wallet].publicKey] = contractInstance
     }
-
-    return `(${ temp.toString() })`;
 }
 
-function parseACIFunctionReturnType (functionReturns : any = []): any {
-    
-    let returnType = functionReturns;
-    if (typeof functionReturns !== 'string') {
-        if (functionReturns.map) {
-            returnType = processReturnType(functionReturns.map);
-        } else if (functionReturns.tuple) {
-            returnType = processReturnType(functionReturns.tuple);
-        } else if (functionReturns.record) {
-            returnType = processReturnTypeRecord(functionReturns.record);
-        } else if (functionReturns.list) {
-            let result = processReturnType(functionReturns.list);
-            result = 'list' + result;
-
-            returnType = result;
-        }
-    }
+async function generateFunctionsFromSmartContract(contractSource: string, deployedContract: DeployedContract, privateKey: number, network: Network, contractInstance: ContractInstance): Promise<Object> {
+    let contractFunctions = contractInstance.methods;
 
-    return returnType;
-}
+    contractFunctions['from'] = async function (userWallet: string) {
```

Comment:
`userWallet: string ` maybe this should be `Any`? because you pass a string and after that you check for a property